### PR TITLE
Support v2 async controllers for posting launch/test/log

### PR DESCRIPTION
**Pull request by nvborisenko**
Hidden stones:
- Finish test request requires specifying of `LaunchUuid`
- Log request requires `LaunchUuid`

### CODE REVIEW

**Comment by nvborisenko** [comment]
Still need to figure out why tests for async logs are failing.

**Comment by nvborisenko** [comment]
Still need to figure out why tests for async logs are failing.

File: src/ReportPortal.Client/Abstractions/Resources/IAsyncLaunchResource.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /// </summary>
+    public interface IAsyncLaunchResource
+    {
+        /// <summary>
+        /// Asynchronously finishes specified launch.
+        /// </summary>
+        /// <param name="uuid">UUID of specified launch.</param>
+        /// <param name="request">Information about representation of launch to finish.</param>
+        /// <returns>A message from service.</returns>
+        Task<LaunchFinishedResponse> FinishAsync(string uuid, FinishLaunchRequest request);
```
Comment (Aleh-Yanushkevich) [inline src/ReportPortal.Client/Abstractions/Resources/IAsyncLaunchResource.cs:19]: Is it really needed for us to have support two method: with and w/o cancellation token? Maybe better just to follow Microsoft async pattern and has only method with cancellation token? For v1 we support both version for backward campatibility.

**Review by Aleh-Yanushkevich — APPROVED** [review]

File: src/ReportPortal.Client/Abstractions/Resources/IAsyncLaunchResource.cs (hunk 1 above)
Comment (nvborisenko) [reply src/ReportPortal.Client/Abstractions/Resources/IAsyncLaunchResource.cs:19]: I even don't mind to leave methods with cancellation tokens only, even for v1, event if it is breaking change. Let's do it in separate PR.

**Pull request merged**